www/js/config/dynamic_config.js

```diff
@@ -138,29 +169,25 @@ angular.module('emission.config.dynamic', ['emission.plugin.logger'])
                 $rootScope.$apply(() => dc.saveAndNotifyConfigReady);
                 return; // labels are the same
             }
-            // if the labels are different
-            return readConfigFromServer(urlComponents.label, urlComponents.source).then((downloadedConfig) => {
-                // we can use angular.extend since urlComponents is not nested
-                // need to change this to angular.merge if that changes
-                const toSaveConfig = angular.extend(downloadedConfig, {joined: urlComponents});
-                const storeConfigPromise = $window.cordova.plugins.BEMUserCache.putRWDocument(
-                    CONFIG_PHONE_UI, toSaveConfig);
-                const logSuccess = (storeResults) => Logger.log("UI_CONFIG: Stored dynamic config successfully, result = "+JSON.stringify(storeResults));
-                // loaded new config, so it is both ready and changed
-                return storeConfigPromise.then(logSuccess)
-                    .then(dc.saveAndNotifyConfigChanged(downloadedConfig))
-                    .then(dc.saveAndNotifyConfigReady(downloadedConfig))
-                    .then($state.go("root.intro"))
-                    .catch((storeError) => Logger.displayError("Error storing downloaded study configuration", storeError));
-            });
+            // if the labels are different, we need to download the new config
+            return loadNewConfig(urlComponents, true);
         });
     };
-    dc.initAtLaunch = function() {
+    dc.initAtLaunch = function () {
         loadSavedConfig().then((existingConfig) => {
-            if(existingConfig) {
-                // the user has already configured the app, let's cache the
-                // config and notify others that we are done
-                Logger.log("UI_CONFIG: finished loading config on app start");
+            if (!existingConfig) {
+                return Logger.log("UI_CONFIG: No existing config, skipping");
+            }
+            // if 'autoRefresh' is set, we will check for updates
+            if (existingConfig.autoRefresh) {
+                loadNewConfig(existingConfig.joined, false, existingConfig.version).then((wasUpdated) => {
```

Comment:
A 'reload' from devtools does the trick for me, or triggering an autoreload from phonegap by saving any file.
To me, this is faster than navigating to the Developer zone would be

When the time comes, it will be easy to change this from a dev feature (wait for the new config to be loaded before use) to a production feature (use the old config this time, but keep checking in background and let updates apply next time) - since it is already written in an asynchronous way